src/Checkout_Blocks/Extend_Block_Core.php

```diff
@@ -273,8 +292,15 @@ public function handle_set_checkout_post_data() {
 		}
 
 		// Check if required fields are present
-		if ( empty( $sanitized_data['shipping_postcode'] ) || empty( $sanitized_data['shipping_house_number'] ) && 'NL' == $shipping_country ) {
-			WC()->session->__unset( 'postnl_checkout_post_data' );
+		if (
+			empty( $sanitized_data['shipping_postcode'] )
+			||
+			(
+				$this->settings->is_reorder_nl_address_enabled()
+				&& empty( $sanitized_data['shipping_house_number'] )
+				&& 'NL' == $shipping_country
+			) ) {
+				WC()->session->__unset( 'postnl_checkout_post_data' );
 			wp_send_json_success(
 				array(
 					'message'          => 'Postcode or house number is missing.',
```

Comment:
this need to be translatable
```suggestion
					'message'          => esc_html__( 'Postcode or house number is missing.', 'postnl-for-woocommerce' ),
```